templates/Test.h.ejs

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "TestCase.h"
+#include <playfab/PlayFabGlobal.h>
+#include <playfab/PlayFabEntity.h>
+#include <playfab/PlayFabClientDataModels.h>
+#include "../../../code/source/Client/<%- api.name %>DataModels.h"
```

Comment:
Not sure I love including the internal header in the test app.  It would be slightly more work, but maybe its worth creating separate C++ versions of the models to release as sample code.  I'm thinking they would look very similar to the internal ones, but perhaps not rely on internal types (BaseModel, PlayFab::String/Vector, etc.).  Seems like they could be helpful for building & cleaning up the request objects compared to using the C structs directly.  Could still include the "FromJson/ToJson" interface since that seems to be good for testing.

Another option would be something like the EventManager::Event class, which essentially makes it easier to build a PlayFabEventsEventContents request.

Reply:
possible.  think about it for a future PR

---
In reply to: [631958540](https://github.com/jplafonta/XPlatCTemplate/pull/16#discussion_r631958540) [](ancestors = 631958540)